packages/flutter/test/cupertino/refresh_test.dart

```diff
@@ -535,6 +528,82 @@ void main() {
       debugDefaultTargetPlatformOverride = null;
     });
 
+    testWidgets('builder still called when sliver snapped back more than 90%', (WidgetTester tester) async {
+      debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+      refreshIndicator = const Center(child: Text('-1'));
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: CustomScrollView(
+            slivers: <Widget>[
+              CupertinoSliverRefreshControl(
+                builder: builder,
+                onRefresh: onRefresh,
+              ),
+              buildAListOfStuff(),
+            ],
+          ),
+        ),
+      );
+
+      await tester.drag(find.text('0'), const Offset(0.0, 150.0));
+      await tester.pump();
+      verify(mockHelper.builder(
+        any,
+        RefreshIndicatorMode.armed,
+        150.0,
+        100.0, // Default value.
+        60.0, // Default value.
+      ));
+      expect(
+        tester.getRect(find.widgetWithText(Center, '-1')),
+        Rect.fromLTRB(0.0, 0.0, 800.0, 150.0),
+      );
+      verify(mockHelper.refreshTask());
+
+      // Rebuilds the sliver with a layout extent now.
+      await tester.pump();
+      // Let it snap back to occupy the indicator's final sliver space only.
+      await tester.pump(const Duration(seconds: 2));
+      verify(mockHelper.builder(
+        any,
+        RefreshIndicatorMode.refresh,
+        60.0,
+        100.0, // Default value.
+        60.0, // Default value.
+      ));
+      expect(
+        tester.getRect(find.widgetWithText(Center, '-1')),
+        Rect.fromLTRB(0.0, 0.0, 800.0, 60.0),
+      );
+      expect(
+        tester.getRect(find.widgetWithText(Center, '0')),
+        Rect.fromLTRB(0.0, 60.0, 800.0, 260.0),
+      );
+
+      refreshCompleter.complete(null);
+      await tester.pump();
+      verify(mockHelper.builder(
+        any,
+        RefreshIndicatorMode.done,
+        60.0,
+        100.0, // Default value.
+        60.0, // Default value.
+      ));
+
+      // Dragging up by more than 90% of sliver height
+      await tester.drag(find.text('0'), const Offset(0.0, 149));
+      expect(find.text('-1'), findsOneWidget);
+      expect(
+        tester.getRect(find.widgetWithText(Center, '0')),
+        Rect.fromLTRB(0.0, 60.0, 800.0, 260.0),
+      );
+
```

Comment:
If you temporarily remove your fix and ran this test again, I don't think it would fail. That's generally a good way to check whether the test works :)

Reply:
Thanks for the feedback, I messed up! Gonna fix it now :)